# Hyperparameter Sweep

Run a bounded sweep, compare validation metrics, and save the ranked results. This tests repeated training, table output, and selection logic.

In [1]:
from pathlib import Path

exec(Path("samples/ml_tasks/hyperparameter_sweep.py").read_text(encoding="utf-8"))


lr=0.030 l2=0.000 epochs=80 f1=0.7586 brier=0.1882
lr=0.030 l2=0.000 epochs=140 f1=0.7586 brier=0.1737
lr=0.030 l2=0.001 epochs=80 f1=0.7586 brier=0.1883
lr=0.030 l2=0.001 epochs=140 f1=0.7586 brier=0.1738
lr=0.030 l2=0.010 epochs=80 f1=0.7586 brier=0.1887
lr=0.030 l2=0.010 epochs=140 f1=0.7586 brier=0.1743
lr=0.060 l2=0.000 epochs=80 f1=0.7586 brier=0.1708
lr=0.060 l2=0.000 epochs=140 f1=0.7619 brier=0.1625
lr=0.060 l2=0.001 epochs=80 f1=0.7586 brier=0.1709
lr=0.060 l2=0.001 epochs=140 f1=0.7619 brier=0.1625
lr=0.060 l2=0.010 epochs=80 f1=0.7619 brier=0.1715
lr=0.060 l2=0.010 epochs=140 f1=0.7619 brier=0.1632
sweep_artifact: artifacts/ml_tasks/sweep_results.json
best_config:
- lr: 0.0600
- l2: 0.0000
- epochs: 140
- f1: 0.7619
- accuracy: 0.7708
- precision: 0.7586
- recall: 0.7652
- brier: 0.1625


lr,l2,epochs,f1,accuracy,brier
0.06,0.0,140,0.762,0.771,0.162
0.06,0.001,140,0.762,0.771,0.163
0.06,0.01,80,0.762,0.771,0.172
0.06,0.01,140,0.762,0.771,0.163
0.03,0.0,80,0.759,0.767,0.188
0.03,0.0,140,0.759,0.767,0.174
0.03,0.001,80,0.759,0.767,0.188
0.03,0.001,140,0.759,0.767,0.174
0.03,0.01,80,0.759,0.767,0.189
0.03,0.01,140,0.759,0.767,0.174


## Notebook Lens CLI Finding

`run-clean` completed successfully in 2.017 seconds and executed 1 of 1 code cells. The best config was `lr=0.0600`, `l2=0.0000`, `epochs=140`, with `f1=0.7619`, `accuracy=0.7708`, `precision=0.7586`, `recall=0.7652`, and `brier=0.1625`.

Artifact produced: `artifacts/ml_tasks/sweep_results.json`.

The CLI made this sweep easy to compare: `state` exposed the stable cell id (`mlsweep1`) and execution status, while `show-cell --outputs full` showed the complete sweep log and the rendered ranked table preview. Output truncation did not get in the way for this notebook.

Workflow feedback: the stdout summary was useful and the artifact path was easy to find. The main confusing part was that the HTML preview is flattened into text, so the ranked table is readable but cramped; a table-aware preview or a first-class artifact summary in `state` would make real sweep comparisons cleaner.
